src/Symfony/Component/Clock/TimePoint.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Clock;
+
+/**
+ * An immmutable DateTime with stricter error handling and return types than the native one.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class TimePoint extends \DateTimeImmutable
```

Comment:
Why extend \DateTimeImmutable?

It seems implementing \DateTimeInterface and using this as a factory to return a \DateTimeImmutable should be sufficient to wrap this constructor.

Reply:
A PSR clock must return a DateTimeImmutable.

And PHP does not let implement DateTimeInterface in userland. It is a special interface (like Traversable for instance)

Reply:
Because the new class must pass `DateTimeImmutable` and `DateTimeInterface` type hints.

Reply:
@stof thank you mentioning the need to honor PSR20—I wasn't aware of that.

Reply:
Well, that's the main responsibility of this Clock component: implementing the ClockInterface.